rosetta/converter/converter.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2021 Alvalor S.A.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package converter
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/onflow/cadence"
+	"github.com/onflow/cadence/encoding/json"
+	"github.com/onflow/flow-go/model/flow"
+
+	"github.com/optakt/flow-dps/models/dps"
+	"github.com/optakt/flow-dps/rosetta/identifier"
+	"github.com/optakt/flow-dps/rosetta/object"
+)
+
+type Converter struct {
+	deposit    flow.EventType
+	withdrawal flow.EventType
+}
+
+func New(gen Generator) (*Converter, error) {
+	deposit, err := gen.TokensDeposited(dps.FlowSymbol)
+	if err != nil {
+		return nil, fmt.Errorf("unable to generate deposit event type: %w", err)
+	}
+	withdrawal, err := gen.TokensWithdrawn(dps.FlowSymbol)
+	if err != nil {
+		return nil, fmt.Errorf("unable to generate withdrawal event type: %w", err)
+	}
```

Comment:
I think it's OK to do it like this for now. However, we might want to support other symbols/currencies in the future, so we might want to generate that for all the tokens that are configured in the DPS parameters.

Reply:
I will add a TODO and an issue for that then, and we'll do it when the time is right